sklearn/exceptions.py

```diff
@@ -80,6 +80,30 @@ class DataDimensionalityWarning(UserWarning):
     the dimensionality of the original source space, to imply that the
     dimensionality of the problem will not be reduced.
 
+
+    Examples:
+    ---------
+    >>> import numpy as np
+    >>> from scipy.sparse import coo_matrix
+    >>> import warnings
+    >>> from sklearn.exceptions import DataDimensionalityWarning
+    >>> from sklearn.random_projection import SparseRandomProjection
+    >>> n_samples, n_features = 10, 1000
+    >>> n_nonzeros = int(n_samples * n_features/100.)
+    >>> rng = np.random.RandomState(0)
+    >>> data_coo = coo_matrix(
+    ...         (rng.randn(n_nonzeros),
+    ...         (rng.randint(n_samples, size=n_nonzeros),
+    ...         rng.randint(n_features, size=n_nonzeros))),
+    ...         shape=(n_samples, n_features)).toarray()
+    >>> with warnings.catch_warnings(record=True) as w:
+    ...     SparseRandomProjection(n_components=n_features + 1).fit(data_coo)
+    ...     print(w[-1].message)
```

Comment:
I'd put the print after the `with ...` construct, check that there's only one there, and print that one.

Reply:
Could you elaborate on this? I see all the other examples follow the same format used here: https://scikit-learn.org/stable/modules/generated/sklearn.exceptions.ConvergenceWarning.html